src/utility.c

```
@@ -0,0 +1,288 @@
+*/
+void KDF(BYTE *k, BYTE *msg, unsigned int msgLength, unsigned int klen)
+{
+	unsigned int ct, v,len2;
+	BYTE *Msg2,hash_byte[32];
+	DWORD hash[8];
+	unsigned int i,n,m,cur;
+	ct = 1;              // Initialize counter to 1
+	v = 256;             // SM3 output is 256 bits
+	len2 = msgLength + 4;   // H = Hv(Z||ct)
+	Msg2 = (BYTE*)malloc(len2);
+	memset(Msg2,0,len2);
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)